packages/core/src/use-cases/locations/delete-locations-use-case.ts

```diff
@@ -0,0 +1,24 @@
+import { NotFoundError } from "../../errors";
+import type { ILocationsRepository } from "../../interfaces";
+
+type Request = {
+    locationsId: string[];
+};
+
+export class DeleteLocationsUseCase {
+    private readonly locationsRepository: ILocationsRepository;
+
+    constructor(locationsRepository: ILocationsRepository) {
+        this.locationsRepository = locationsRepository;
+    }
+
+    async execute({ locationsId }: Request): Promise<void> {
+        for (const locationId of locationsId) {
+            const existingLocation = await this.locationsRepository.findById(locationId);
+            if (!existingLocation) {
+                throw new NotFoundError(`Local com ID ${locationId} não encontrado`);
```

Comment:
Id is not necessary in the error message., as the user does not  know what an id is.